### PR TITLE
[PoC] Check DELETE expectations for mocks 1178

### PR DESCRIPTION
**Pull request by mapedraza**
This PR is created with the aim of modify the expectations to (this should break the tests because in some cases the DELETE is sending some data as described in #1178)

### CODE REVIEW

**Comment by mapedraza** [comment]
Now the test edited ins failing as shown in [tests results log](https://github.com/telefonicaid/iotagent-node-lib/runs/4855922694?check_suite_focus=true#step:5:1626) because it is receiving `"body": "true"`

>  message: 'Nock: No match for request {\n  "method": "DELETE",\n  "url": "http://192.168.1.1:1026/v2/registrations/6319a7f5254b05844116584d",\n  "headers": {\n    "user-agent": "got (https://github.com/sindresorhus/got)",\n    "fiware-service": "smartgondor",\n    "fiware-servicepath": "gardens",\n    "accept": "application/json",\n    "content-type": "application/json",\n    "content-length": "4",\n    "accept-encoding": "gzip, deflate, br"\n  },\n  **"body": "true"\n**}'

**Comment by mapedraza** [comment]
In order to check if expectations would work if the request is not sending data, next step is to remove the body, fixing `json : true` in the code as needed in #1178

**Comment by mapedraza** [comment]
patch generated from #1178 applied to this branch. Now, the edited test should not fail

File: lib/services/devices/registrationUtils.js [hunk 1]
```diff
@@ -136,7 +136,6 @@ function sendUnregistrationsNgsi2(deviceData, callback) {
     const options = {
         url: cbHost + '/v2/registrations/' + deviceData.registrationId,
         method: 'DELETE',
-        json: true,
```
Comment (AlvaroVega) [inline lib/services/devices/registrationUtils.js:139]: This was already deleted by https://github.com/telefonicaid/iotagent-node-lib/pull/1178

Comment (mapedraza) [reply lib/services/devices/registrationUtils.js:139]: This PR is a PoC. Changes in #1178 were included here in order to check if changes on tests were capturing the problem exposed and, after applying the patch, if it continues or not.

File: test/unit/ngsiv2/provisioning/device-registration_test.js [hunk 2]
```diff
@@ -290,7 +290,7 @@ describe('NGSI-v2 - IoT Agent Device Registration', function () {
             contextBrokerMock.post('/v2/entities?options=upsert').reply(204);
 
             contextBrokerMock
-                .delete('/v2/registrations/6319a7f5254b05844116584d')
+                .delete('/v2/registrations/6319a7f5254b05844116584d',"")
```
Comment (mapedraza) [inline test/unit/ngsiv2/provisioning/device-registration_test.js:293]: The important change in this PR is this change, which checks the body of the DELETE request (in this particular case, checks if it is empty)

**Comment by mapedraza** [comment]
After creating https://github.com/telefonicaid/iotagent-node-lib/pull/1183, this PR can be closed since the changes needed to enhance the expectations were made in https://github.com/telefonicaid/iotagent-node-lib/pull/1183

**Pull request closed without merging**